platform/android/src/android_gl_thread.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <mbgl/actor/actor.hpp>
+#include <mbgl/actor/mailbox.hpp>
+#include <mbgl/actor/scheduler.hpp>
+#include <mbgl/renderer/renderer.hpp>
+
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <utility>
+
+namespace mbgl {
+namespace android {
+
+class AndroidGLThread : public Scheduler {
```

Comment:
`AndroidGLThread` should also have direct access to the `GLSurfaceView` object. Implement `schedule` by invoking `queueEvent` via JNI, passing a new instance of a Java class which:

* Implements `Runnable`
* Has a native peer
* Native peer has a `std::weak_ptr<Mailbox> mailbox` member
* Implements `run` using native method that calls `Mailbox::maybeReceive(mailbox)`

Then you can remove `AndroidGLThread::process`, `AndroidGLThread::mutex`, `AndroidGLThread::queue`, `InvalidateCallback`, `NativeMapView::requestProcessing`, etc.